Enables an option to allow clients to be found for any domain in the certification path OHM-114

rcrichton opened the pull request:
Closes #418 

@devcritter could you please review this for me? @bausmeier I'm not sure if you want to take a look as well.

I'm not too happy about having a dependency on the [pem](https://github.com/andris9/pem) modules master branch. I wonder if we should wait for a release to merge this in. Thoughts?

devcritter commented on README.md:
```diff
@@ -77,11 +77,12 @@ This starts the server with production defaults, including the use of the produc
 * `grunt watch` - will automatically build the project on any changes.
 * `grunt lint` - ensure the code is lint free, this is also run before an `npm test`
 * `npm link` - will symlink you local working directory to the globally installed openhim-core module. Use this so you can use the global openhim-core binary to run your current work in progress. Also, if you build any local changes the server will automatically restart.
+* `grunt test --mochaGrep=<regex>` - will only run tests with names matching the regex
```
nice!

devcritter commented:
:+1: Yeah, looks good to me. It took me a bit to fully understand; but it makes sense now. The `clientLookup` method is really elegant, nice one!

On the pem module master branch: I'm actually okay with having that in master. I would rather just say that we should wait before doing a release.

rcrichton commented:
Thanks @devcritter. It was rather confusing to write to! :) - I've added some better comments.

I've added a few fixes and better testing from some of the issue @bausmeier chatted to me about.

Could you guys have a quick look again?

devcritter commented:
:+1: Looks good to me

bausmeier commented:
:+1:
